### PR TITLE
V22F-453 updated IPCLib

### PR DESCRIPTION
**Pull request by KungFuDonkey**

### CODE REVIEW

File: SDALib-CPP/SDAReplay/ReplayDriver.h [hunk 1]
```diff
@@ -73,8 +73,8 @@ class ReplayDriver : public SDADriver
 
         if (!m_replayFile.is_open() || p_data.TickCount == MAX_ULONG) return action;
 
-        std::cout << "TickCount: " << p_data.TickCount << "\n"
-                  << "Tick To Act: " << m_tickToAct << std::endl;
+        // std::cout << "TickCount: " << p_data.TickCount << "\n";
+        //           << "Tick To Act: " << m_tickToAct << std::endl;
```
Comment (Sindarin27) [inline SDALib-CPP/SDAReplay/ReplayDriver.h:77]: Commented code should probably be removed

File: SDALib-CPP/SDAReplay/ReplayDriver.h [hunk 2]
```diff
@@ -102,10 +102,10 @@ class ReplayDriver : public SDADriver
         READ_FLOAT(floatToRead)
         action.Gear = (int)floatToRead;
 
-        std::cout << "\tSteer: " << action.Steer << "\n\t"
+        /*std::cout << "\tSteer: " << action.Steer << "\n\t"
                   << "Brake: " << action.Brake << "\n\t"
                   << "Accel: " << action.Accel << "\n\t"
-                  << "Gear: " << action.Gear << std::endl;
+                  << "Gear: " << action.Gear << std::endl;*/
```
Comment (Sindarin27) [inline SDALib-CPP/SDAReplay/ReplayDriver.h:108]: Commented code should probably be removed.

File: SDALib-CPP/SDALib-CPP/SDADriver.hpp [hunk 3]
```diff
@@ -55,11 +61,18 @@ class SDALIB_EXPORT AIInterface
...
     bool Update()
     {
+        SDA_LOG_INFO("Updating...");
         const int err = m_client.AwaitData(m_buffer, SDA_BUFFER_SIZE);  // can change to GetData
+        SDA_LOG_INFO("Received data");
 
-        if (err != IPCLIB_SUCCEED) return false;
+        if (err != IPCLIB_SUCCEED)
+        {
+            std::cout << "IPCLIB_FAIL" << std::endl;
```
Comment (Sindarin27) [inline SDALib-CPP/SDALib-CPP/SDADriver.hpp:70]: Can you write which kind of fail? Or is there only one kind of fail?

**Review by Sindarin27 — CHANGES_REQUESTED** [review]

File: SDALib-CPP/SDAReplay/ReplayDriver.h [hunk 4]
```diff
@@ -4,10 +4,30 @@
...
 
+#ifdef LOG_REPLAY
+
+#define LOG_VALUES()                                   \
+    std::cout << "\tSteer: " << action.Steer << "\n\t" \
+              << "Brake: " << action.Brake << "\n\t"   \
+              << "Accel: " << action.Accel << "\n\t"   \
+              << "Gear: " << action.Gear << std::endl
+
+#define LOG_TICK()                                          \
```
Comment (Sindarin27) [inline SDALib-CPP/SDAReplay/ReplayDriver.h:20]: LOG_TICK seems to be unused

**Review by Sindarin27 — CHANGES_REQUESTED** [review]

**Review by klieverse — APPROVED** [review]

**Review by Sindarin27 — APPROVED** [review]

**Pull request merged**